_posts/2016-10-08-jumping-into-elasticsearch.markdown

```diff
@@ -0,0 +1,348 @@
+
+### Compound queries
+All of the queries described so far can be combined into [compound queries](https://www.elastic.co/guide/en/elasticsearch/reference/current/compound-queries.html). This is where things really start to pick up, but we'll continue to stay pretty basic.
+
+The `bool` query uses one or more boolean clauses—`must`, `must_not`, `should`, and `filter`—each with one or more nested queries (including potentially *more* `bool` queries, with *their own* clauses and queries), to determine matching results. The `must` and `filter` clauses are very similar, with the only difference being that `filter` ignores the query's score. The `should` clause contributes to the document's score if the nested query is satisfied, but has no effect otherwise.
+
+```
+GET /socialnetwork/user/_search
+{
+  "query": {
+    "bool": {
+      "should": [
+        { 
+          "range" : {
+            "age" : {
+              "gte" : 26,
+              "lte" : 35
+            }
+          }
+        },
+        { "term": { "has_pets": true } }
+      ],
+      "must": { "match": { "state": "New York" } },
+      "must_not": {
+        "range": {
+          "age": { "gt": 50 }
+        }
+      }
+    }
+  }
+}
+```
+
+Remember at the start of this Searching section I mentioned there was a `localsearch` index available? Let's dust that off.
+
+The `indices` query allows for searches across multiple indexes, similar to what Netflix does when including actors along with titles in their search suggestions. Here's a simplistic example expanding upon the previous search:
+
+```
+GET /_search
+{
+  "query": {
+    "indices": {
+      "indices": [ "socialnetwork", "localsearch" ],
+      "query": {
+        "bool": {
+          "should": [
+            {
+              "multi_match": {
+                "fields": [ "skills", "tags" ],
+                "query": "Algorithms"
+              }
+            },
+            {
+              "range": {
+                "age": {
+                  "gte": 26,
+                  "lte": 35
+                }
+              }
+            }
+          ],
+          "must": { "match_phrase": { "state": "New York" } },
+          "must_not": [
+            {
+              "range": {
+                "age": { "gt": 50 }
+              }
+            },
+            { "match": { "skills": "Art Exhibitions" } }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+Here we're looking for users that are ideally skilled in Algorithms and somewhere between 26 and 35 years old (`should`), definitely located in New York state (`must`), and defintely *not* older than 50 or skilled in Art Exhibitions (`must_not`). (Gotta draw the line somewhere. Thanks, mock data.) At the same time, we're looking for businesses that are also in New York, and ideally have `'Algorithms'` in their tags.
```

Comment:
>Gotta draw the line somewhere

xD

Hacker News can drive the best traffic but gets on an off-topic hate train _quickly_. One potential pitfall here is that there's a lot of upheaval currently about Google getting sued by engineers over the age of 40 for being wrongfully discriminated against during the hiring process. Age-blocking this might be a hot-button. It could be a reach, though, so it's your call :)